clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -73,7 +73,7 @@ class Util {
   template <size_t N>
   static constexpr DeclContextDesc MakeDeclContextDesc(Decl::Kind K,
                                                        const char (&Str)[N]) {
-    return DeclContextDesc{K, llvm::StringLiteral{Str}};
+    return DeclContextDesc{K, llvm::StringRef{Str, N}};
```

Comment:
Can you add a FIXME comment here explaining a bit about why we're doing this funny dance? I worry someone later will revert back to the original code, otherwise.